src/wp-admin/includes/image.php

```diff
@@ -301,18 +310,42 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 			if ( ! is_wp_error( $resized ) ) {
 				// Append "-scaled" to the image file name. It will look like "my_image-scaled.jpg".
 				// This doesn't affect the sub-sizes names as they are generated from the original image (for best quality).
-				$saved = $editor->save( $editor->generate_filename( 'scaled' ) );
+				$saved = $editor->save( $editor->generate_filename( 'scaled' ), $primary_mime_type );
 
 				if ( ! is_wp_error( $saved ) ) {
 					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
 
+					$image_meta['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $saved );
+
 					// If the image was rotated update the stored EXIF data.
 					if ( true === $rotated && ! empty( $image_meta['image_meta']['orientation'] ) ) {
 						$image_meta['image_meta']['orientation'] = 1;
 					}
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
 				} else {
 					// TODO: Log errors.
 				}
+
+				// Save additional mime types.
+				foreach ( $additional_mime_types as $output_mime_type ) {
+					$editor = wp_get_image_editor( $file );
+
+					if ( is_wp_error( $editor ) ) {
+						// This image cannot be edited.
+						continue;
+					}
```

Comment:
See https://github.com/WordPress/wordpress-develop/pull/2393#discussion_r831609180, I'm still doubtful on the need for this. We implemented it like that in the plugin I believe, but looking at the WP core implementation, it uses the same editor instance to generate all versions - so why should we do it differently here? That doesn't make sense to me.

Again, I realize we discussed in the plugin implementation that we want to use the "original" image. Are we sure we need a new editor instance to do that? Particularly because WP core doesn't do that.

On another note, you're overriding the `$editor` local variable here, which will also affect subsequent cycles of the outer `foreach` loop, precisely _because_ WordPress core's logic is built to use a single editor instance. Also, if you wanted to reinstantiate the editor here, you'd also need to reapply any orientation and big image cropping from before, which is currently missing - another reason I think we shouldn't do this.

Reply:
👍🏼  removed